libopenage/game_control.cpp

```diff
@@ -562,6 +568,9 @@ void GameControl::set_mode(int mode_index) {
 			engine->get_input_manager().remove_context(this->active_mode);
 
 			// set the new active mode
+			if (this->active_mode) {
```

Comment:
Just for future error-avoidance, can you change this to a check to test if the previous mode is not the new mode?
that way, a non-change in modes doesn't call the exit. This should probably be done up top in that function.